Omit `never` metadata properties in openapi3 and document for other emitter/users (#1338)
microsoft/cadl: docs/language-basics/models.md

}
```

### Special property types

#### `never`

> A record property can be declared as having the type never. This should mean that the record shouldn't be having that property.

Reviewer: ```suggestion
A model property can be declared as having the type never. This can be interpreted as the model not having that property.
```

> This can be useful in model template to omit a property.

Reviewer: ```suggestion
This can be useful in a model template to omit a property.
```

```cadl
model Address<TState> {
  state: TState;
  city: string;
  street: string;
}

model UKAddress is Address<never>;
```

:::note

> That is still up to the emitter to be removing `never` properties. Cadl compiler will not automatically omit those.

Reviewer: ```suggestion
It is up to the emitter to remove `never` properties. The Cadl compiler will not automatically omit them.
```
